---
float64 position
int32 exception_code
